### PR TITLE
feat:  Add digital signature support

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
Dependent on:  https://github.com/hpcc-systems/Visualization/pull/3745

File: src/ecl/command.ts [hunk 1]
```diff
@@ -134,4 +135,36 @@ export class ECLCommands {
...
         }
     }
+
+    sign() {
+        if (vscode.window.activeTextEditor && sessionManager.session) {
+            const editor = vscode.window.activeTextEditor;
+            const eclText = editor.document.getText();
+            sessionManager.session.digitalKeys().then(async keys => {
+                if (keys.length === 0) {
+                    vscode.window.showWarningMessage(localize("Environment has no digital keys"));
```
Comment (shamser) [inline src/ecl/command.ts:145]: I think it would be clearer if it was called: code signing key

File: src/ecl/command.ts [hunk 2]
```diff
@@ -134,4 +135,36 @@ export class ECLCommands {
...
+                } else {
+                    const key = await vscode.window.showQuickPick(keys, {
+                        placeHolder: localize("Select Digital Key")
```
Comment (shamser) [inline src/ecl/command.ts:148]: Select code signing key?

File: src/ecl/command.ts [hunk 3]
```diff
@@ -134,4 +135,36 @@ export class ECLCommands {
...
+                    });
+                    if (key) {
+                        const passphrase = await vscode.window.showInputBox({
+                            prompt: localize("Enter passphrase for") + ` "${key}"`
```
Comment (shamser) [inline src/ecl/command.ts:152]: Will the user need to select the key and enter the passphrase every time they sign the code  or will it be cached?

**Review by shamser — CHANGES_REQUESTED** [review]
@GordonSmith  Couple of minor suggestions.

File: src/ecl/command.ts (hunk 3 above)
Comment (GordonSmith) [reply src/ecl/command.ts:152]: Everytime - but it is a picklist.

File: src/ecl/command.ts (hunk 1 above)
Comment (GordonSmith) [reply src/ecl/command.ts]: Agree

File: src/ecl/command.ts (hunk 2 above)
Comment (GordonSmith) [reply src/ecl/command.ts]: Changed

File: src/ecl/command.ts (hunk 3 above)
Comment (GordonSmith) [reply src/ecl/command.ts:152]: (Also folks can switch environments easily... )

**Pull request merged**